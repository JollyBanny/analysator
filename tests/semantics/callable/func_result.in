function doSomething(): integer;
begin
    result := 20;
end;

var a: integer;

begin
    a:=doSomething();
end.