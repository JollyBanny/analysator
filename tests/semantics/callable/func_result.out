program
└──── program block
      ├──── FUNCTION
      │     ├──── header
      │     │     ├──── dosomething
      │     │     └──── integer
      │     └──── block
      │           └──── Compound statement
      │                 └──── :=
      │                       ├──── result
      │                       └──── 20
      ├──── VAR
      │     └──── var declaration
      │           ├──── a
      │           └──── integer
      └──── Compound statement
            └──── :=
                  ├──── a
                  └──── dosomething

┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│  builtins                                                                                                            │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  integer                                                                                         │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  double                                                                                          │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  char                                                                                            │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  string                                                                                          │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  boolean                                                                                         │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│  global                                                                                                              │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Function         │  dosomething                │  integer                                                           │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
    ┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
    │  dosomething params:                                                                                                 │
    ├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
    ┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
    │  dosomething locals:                                                                                                 │
    ├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
    │  Var              │  result                     │  integer                                                           │
    └──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
│  Var              │  a                          │  integer                                                           │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
